framework/include/restart/Restartable.h

```diff
@@ -211,6 +211,9 @@ class Restartable
   friend class Transient;
   friend class TableOutput;
   friend class TransientMultiApp;
+
+  /// For access to declareRestartableData() while avoiding diamond inheritance
+  friend class MutableCoefficientsInterface;
```

Comment:
I would rather move `declareRestartableData()` to being public.  What do you think @permcody ?

Reply:
Yes, but I'm also OK, if we put this in now. I looked at this briefly this morning and it'll be easy to remove this after the fact. I'm willing to fix this up myself. Valgrind tests have passed.